src/amo/components/RatingsByStar/index.js

```diff
@@ -143,13 +143,21 @@ export class RatingsByStarBase extends React.Component<InternalProps> {
                   )}
                   <IconStar selected />
                 </div>
-                <div className="RatingsByStar-barContainer">
-                  <div className="RatingsByStar-bar RatingsByStar-barFrame">
-                    {starCount !== undefined
-                      ? this.renderBarValue(starCount)
-                      : null}
+                {loading ? (
+                  <div className="RatingsByStar-barContainer">
+                    <div className="RatingsByStar-bar RatingsByStar-barFrame" />
                   </div>
-                </div>
+                ) : (
+                  createLink(
+                    <div className="RatingsByStar-barContainer">
```

Comment:
Would it be possible to call `createLink()` inside the `.RatingsByStar-barContainer` component?

```js
<div className="RatingsByStar-barContainer">
  {createLink(
    <div className="RatingsByStar-bar RatingsByStar-barFrame">
       ...
    </div>
  )}
</div>
```

I think this would help writing more concise test cases.

For instance, in loading state, we could make sure no `Link` are rendered. In non-loading state, we could `.find(Link)` inside `.RatingsByStar-barContainer`, which means writing a test case very similar to `it('renders star labels and links', () => {});`. I even think we can reuse it and use `it.each()`.

Reply:
Great. It works and makes it much easier to writing test cases.